src/components/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { UserWarning } from '../../UserWarning';
+import * as todosService from '../../api/todos';
+
+const USER_ID = 11383;
+
+type Props = {
+  children: React.ReactNode;
+};
+
+type TodoContextValue = {
+  todos: Todo[];
+  todosUncompleted: number;
+  todosCompleted: Todo[];
+  addTodo: (inputValue: string) => void;
+  toggleTodo: (id: number) => void;
+  toogleAll: () => void;
+  deleteTodo: (id: number) => void;
+  deleteComplitedTodo: () => void;
+  updateTodo: (updatedTitle: string, id: number) => void;
+  isError: boolean;
+  setIsError: (isError: boolean) => void;
+  errorMessage: string;
+  setErrorMessage: (errorMessage: string) => void;
+  tempoTodo: Todo | null;
+  setTempoTodo: (tempoTodo: Todo | null) => void;
+  inputValue: string;
+  setInputValue: (inputValue: string) => void;
+  isOnAdd: boolean;
+  setIsOnAdd: (isOnAdd: boolean) => void;
+  isCompliteDeleting: boolean;
+};
+
+export const TodoContext = React.createContext<TodoContextValue>({
+  todos: [],
+  todosUncompleted: 0,
+  todosCompleted: [],
+  addTodo: () => {},
+  toggleTodo: () => {},
+  toogleAll: () => {},
+  deleteTodo: () => { },
+  deleteComplitedTodo: () => { },
+  updateTodo: () => { },
+  isError: false,
+  setIsError: () => {},
+  errorMessage: '',
+  setErrorMessage: () => {},
+  tempoTodo: null,
+  setTempoTodo: () => {},
+  inputValue: '',
+  setInputValue: () => {},
+  isOnAdd: false,
+  setIsOnAdd: () => {},
+  isCompliteDeleting: false,
+});
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempoTodo, setTempoTodo] = useState<Todo | null>(null);
+  const [isOnAdd, setIsOnAdd] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [isCompliteDeleting, setIsCompliteDeleting] = useState(false);
+
+  useEffect(() => {
+    if (USER_ID) {
+      todosService.getTodos(USER_ID)
+        .then((fetchedTodos) => {
+          setTodos(fetchedTodos);
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error(error);
+        });
+    }
+  }, []);
+
+  const todosUncompleted = useMemo(() => todos.filter(
+    todo => !todo.completed,
+  ).length, [todos]);
+
+  const todosCompleted = useMemo(() => todos.filter(
+    todo => todo.completed,
+  ), [todos]);
+
+  const addTodo = async () => {
+    let newTodo: Todo | null = null;
```

Comment:
function `addTodo` it too large, it's better to divide it for separate functions